### ISSUE TITLE
opam 2.2.0~alpha2: OPAMSWITCH does not work anymore

### ISSUE THREAD

**Issue by eponier** [ocaml/opam#5696]
The reason is that the shell hook resets `OPAMSWITCH` to the empty string as soon as it is set. This is because `opam env --shell=zsh --readonly` (my shell is zsh) prints `OPAMSWITCH=''; export OPAMSWITCH;` as the first line.

Before, the same command did not print `OPAMSWITCH`. This seems to be a consequence of https://github.com/ocaml/opam/pull/5417.

**Comment by eponier** [comment]
I turned off the shell hook to experiment. By default, `opam env --shell=zsh --readonly 2> /dev/null <&-` does not mention `OPAMSWITCH`. But if I type `eval $(opam env --set-switch --switch switch2)` (while global switch is `switch1`), `opam env --shell=zsh --readonly 2> /dev/null <&-` contains as the first line `OPAMSWITCH=''; export OPAMSWITCH;`.
If I just set `OPAMSWITCH` manually, this does not happen (but then I have an unsynchronized environment, and, strangely, this not not reported by opam).
